### ISSUE TITLE
Unlisted versions of Recommended Extensions are being signed as Recommended

### ISSUE THREAD

**Issue by jvillalobos** [mozilla/addons-server#12722]
It looks like all unlisted versions of a listed Recommended add-on are being signed as if they are Recommended. For example, unlisted versions of NoScript since June 24, 2019 all appear to have the Recommended file in them. Since those files aren't being reviewed, they shouldn't have the Recommended file.

**Comment by diox** [comment]
(In its current state, the pull request for https://github.com/mozilla/addons-server/issues/12622 would fix it... sort of: it would make them require manual approval... Something to consider when we are finally able to merge it)

**Comment by jvillalobos** [comment]
Yeah, that wouldn't be good. Those versions should still be auto-approved.

The only "supported" process we have right now for Recommended unlisted versions is that the Recommended file is already in the uploaded API and the file is already signed by our privileged Mozilla signature. I don't expect us to need anything different from that in the immediate future.

**Comment by wagnerand** [comment]
Note that we might have to change the code again once we want to start recommending self-hosted add-ons.

Should the recommendation flag be per channel instead, e.g. having two checkboxes in the "recommendations" admin that let's you decide which channel we should consider recommended (and then disabled auto-approval for selected channels)?

**Comment by jvillalobos** [comment]
I don't think we need that level of complexity right now. Use cases for unlisted recommended extensions are very slim as it is.

**Comment by AlexandraMoga** [comment]
Verified fixed on -dev.

I've submitted an unlisted version for a recommended extension using both devhub and the signing API and the unlisted version will not contain the `mozilla-recommendation.json` file after auto-signing.

An example: 
https://addons-dev.allizom.org/en-US/developers/addon/donottrackplus/versions/1689598
Files in package after auto-signing:
![image](https://user-images.githubusercontent.com/31961530/67692762-56a5b380-f9a9-11e9-85c7-f192686c2004.png)

New listed versions are still created with the recommendation file.
